### PR TITLE
Restructuring Socket/TCP conceptual docs PR

### LINKED ISSUE #31702: Restructuring Socket/TCP conceptual docs
**Describe the tasks**

In an offline discussion with @antonfirsov @dotnet/ncl team, we have decided to do following tasks on conceptual docs:

- [ ] Repurpose `Use Sockets to send and receive data` as `Use Sockets to send and receive data over TCP`.
- [ ] Merge all content under `TCP` as `Use TcpClient and TcpListener` and move it under the `Sockets`.
- [ ] Add a remark to `Use TcpClient and TcpListener`, that we recommend using `Socket` class directly, instead of it.
- [ ] Create an article for UDP (`Use Sockets to send and receive data over UDP`) (???)
- [ ] Create an article to show `TcpClient` usages equivalent to `Socket` under `Use TcpClient and TcpListener`.

All other ideas and feedback are appreciated. 

CC @dotnet/docs @IEvangelist

### PR DESCRIPTION
**Pull request by liveans**
## Summary

Working on it, just creating as a draft to see previews.

Fixes 3 tasks of #31702 

Preview URL: https://review.learn.microsoft.com/en-us/dotnet/fundamentals/networking/sockets/tcp-classes?branch=pr-en-us-32022

### CODE REVIEW

File: docs/fundamentals/networking/sockets/tcp.md
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/tcp.md:31]:
```suggestion
> <xref:System.Net.Sockets.Socket> class is highly recommended for advanced users, instead of `TcpClient` and `TcpListener`.
```

File: docs/fundamentals/networking/sockets/tcp.md
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/tcp.md:33]:
```suggestion
## Use TcpClient and TcpListener
```

Fix build error (MD linter)

File: docs/fundamentals/networking/sockets/tcp.md
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/tcp.md:81]:
```suggestion
### `TcpListener` and `TcpClient` usage equivalent of `Socket`
```

**Review by IEvangelist — COMMENTED** [review]
Looks good thus far, we'll need to have two new redirect entries that point to the _tcp.md_ article for these two deleted files.

**Comment by IEvangelist** [comment]
> > Looks good thus far, we'll need to have two new redirect entries that point to the _tcp.md_ article for these two deleted files.
> 
> Thanks for the prompt review and comments @IEvangelist! I'll update it accordingly.

No worries, are you familiar with how to update the redirects?

**Comment by liveans** [comment]
> > > Looks good thus far, we'll need to have two new redirect entries that point to the _tcp.md_ article for these two deleted files.
> > 
> > 
> > Thanks for the prompt review and comments @IEvangelist! I'll update it accordingly.
> 
> No worries, are you familiar with how to update the redirects?

Not really, I'm trying to understand it.

**Comment by IEvangelist** [comment]
> > > > Looks good thus far, we'll need to have two new redirect entries that point to the _tcp.md_ article for these two deleted files.
> > > 
> > > 
> > > Thanks for the prompt review and comments @IEvangelist! I'll update it accordingly.
> > 
> > 
> > No worries, are you familiar with how to update the redirects?
> 
> Not really, I'm trying to understand it.

If you'd like, I can help you with that. No worries, otherwise, if you're interested, I could teach you - just let me know.

**Comment by IEvangelist** [comment]
> > > > > Looks good thus far, we'll need to have two new redirect entries that point to the _tcp.md_ article for these two deleted files.
> > > > 
> > > > 
> > > > Thanks for the prompt review and comments @IEvangelist! I'll update it accordingly.
> > > 
> > > 
> > > No worries, are you familiar with how to update the redirects?
> > 
> > 
> > Not really, I'm trying to understand it.
> 
> If you'd like, I can help you with that. No worries, otherwise, if you're interested, I could teach you - just let me know.

Have a look at the _https://github.com/dotnet/docs/blob/main/.openpublishing.redirection.core.json_ file. You'll need two new entries, one for each of the deleted files in this PR, both pointing to the TCP route.

File: docs/fundamentals/networking/sockets/tcp.md
Comment (antonfirsov) [inline docs/fundamentals/networking/sockets/tcp.md:24]: This is not the only, and not necessarily the preferred way to work with TCP. We should rephrase this intro in a way that makes it clear that `TcpClient` and `TcpListener` are only helper classes for ease of use.
I would even rename the file from `tcp.md` to `tcp-classes.md`.

File: docs/fundamentals/networking/sockets/tcp.md
Comment (antonfirsov) [inline docs/fundamentals/networking/sockets/tcp.md:31]: This shouldn't be at the end. Instead of presenting this information in a highlighted section, I would consider making this an organic part of the intro paragraph (see my previous comment).

**Review by antonfirsov — COMMENTED** [review]
@IEvangelist is there a way to render the resulting docs in a way done in `dotnet-api-docs` PR-s? Would be nice to see the look & feel of the resulting structure.

**Comment by IEvangelist** [comment]
> @IEvangelist is there a way to render the resulting docs in a way done in `dotnet-api-docs` PR-s? Would be nice to see the look & feel of the resulting structure.

Hi @antonfirsov - yes, if you expand the build details for the OpenPublishing.Build section, you'll see the preview. We usually manually add them to a PR:

https://review.learn.microsoft.com/en-us/dotnet/fundamentals/networking/sockets/tcp?branch=pr-en-us-32022

File: docs/fundamentals/networking/sockets/tcp.md
Comment (liveans) [reply docs/fundamentals/networking/sockets/tcp.md:31]: Yeah, I agree with your previous comment, but I think having a highlighted section would be nice for this. I can move it to start of the page. Thoughts?

File: docs/fundamentals/networking/sockets/tcp-classes.md
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/tcp-classes.md:2]:
```suggestion
title: Use TcpClient and TcpListener
```
We already have a ".NET" suffix added.

File: docs/fundamentals/networking/sockets/tcp-classes.md
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/tcp-classes.md:22]:
```suggestion
# TCP overview
```

We follow sentence casing for headings

File: docs/fundamentals/networking/sockets/tcp-classes.md
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/tcp-classes.md:33]:
```suggestion
## Use `TcpClient` and `TcpListener`
```

File: docs/fundamentals/networking/sockets/tcp-classes.md
Comment (antonfirsov) [inline docs/fundamentals/networking/sockets/tcp-classes.md:27]:
```suggestion
To work with Transmission Control Protocol (TCP), you have two options: either use <xref:System.Net.Sockets.Socket> for maximum control and performance, or use the <xref:System.Net.Sockets.TcpClient> and <xref:System.Net.Sockets.TcpListener> helper classes. <xref:System.Net.Sockets.TcpClient> and <xref:System.Net.Sockets.TcpListener> are built on top of the <xref:System.Net.Sockets.Socket?displayProperty=nameWithType> class and take care of the details of transferring data for ease of use.
```

**Review by antonfirsov — APPROVED** [review]
A few suggestions to improve wording, otherwise LGTM.

**Comment by liveans** [comment]
I'll merge this after @IEvangelist's approve or if you approve, you can merge as well @IEvangelist!

**Review by antonfirsov — APPROVED** [review]
:shipit:

**Pull request merged**
